tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_selectors.ts

```diff
@@ -461,6 +465,49 @@ export const getFocusedExecutionData = createSelector(
   }
 );
 
+/**
+ * Get information regarding the op that's the origin of the focused
+ * code location (stack trace).
+ * This selector covers both eager execution and graph-op creation.
+ */
+export const getCodeLocationOrigin = createSelector(
+  selectDebuggerState,
+  getFocusedExecutionIndex,
+  getFocusedExecutionData,
+  getFocusedGraphOpInfo,
+  (
+    state: DebuggerState,
+    executionIndex: number | null,
+    executionData: Execution | null,
+    graphOpInfo: GraphOpInfo | null
+  ): CodeLocationOrigin | null => {
```

Comment:
Technically, `CodeLocationOrigin` is the wrong type. You want `CodeLocationExecutionOrigin | CodeLocationGraphOpCreationOrigin | null`.

i.e., because the typing is wrong, you have to cast it to `CodeLocationExecutionOrigin` in stack_trace_container.ts.

Reply:
Good catch! Thanks for pointing this out. By using the correct union typing, there is no longer need for casting (see latest commit).